src/main/java/core/basesyntax/service/impl/DataConverterImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.DataConverter;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convert(List<String> lines) {
+        return lines.stream()
+                .map(this::convertToFruitTransaction)
+                .toList();
+    }
+
+    private FruitTransaction convertToFruitTransaction(String line) {
+        String[] fields = line.split(",");

```

Comment:
Consider handling the potential ArrayIndexOutOfBoundsException that can occur if the line does not have enough comma-separated values. This is part of input validation.